src/operator/time_series/rolling_aggregate.rs

```diff
@@ -391,23 +390,21 @@ where
                     // println!("tree: {treestr}");
                     // tree_partition_cursor.rewind_keys();
 
-                    let agg = tree_partition_cursor.aggregate_range::<S>(&range);
-                    // println!("key: {:?}, range: {:?}, agg: {:?}", input_range_cursor.key(),
-                    // range, agg);
-
                     while input_range_cursor.val_valid() {
                         // Generate output update.
                         if !input_range_cursor.weight().le0() {
+                            let agg = tree_partition_cursor.aggregate_range::<S>(&range);
+                            // println!("key: {:?}, range: {:?}, agg: {:?}",
+                            // input_range_cursor.key(), range, agg);
+
                             insertion_builder.push((
                                 O::item_from(
                                     delta_cursor.key().clone(),
-                                    (
-                                        *input_range_cursor.key(),
-                                        (input_range_cursor.val().clone(), agg.clone()),
-                                    ),
+                                    (*input_range_cursor.key(), agg),
                                 ),
                                 HasOne::one(),
                             ));
+                            break;
```

Comment:
I find this surprising

Reply:
As soon as we encounter an input record with non-zero weight, we know that we need to compute the sliding window aggregate for the current timestamp.  We don't care about the actual value or the number of different values, so we can break out of the loop.